communication/nats_dialog/dialog_establisher.go

```diff
@@ -28,40 +30,48 @@ const establisherLogPrefix = "[NATS.DialogEstablisher] "
 
 type dialogEstablisher struct {
 	myIdentity            identity.Identity
+	mySigner              identity.Signer
 	contactAddressFactory func(contact dto_discovery.Contact) (*nats_discovery.NatsAddress, error)
 }
 
 func (establisher *dialogEstablisher) CreateDialog(contact dto_discovery.Contact) (communication.Dialog, error) {
+	var dialog *dialog
```

Comment:
Adding this variable at the beginning of method and using it before initializing does not improve readability IMO:
```go
var dialog *dialog
...
return dialog, fmt.Errorf("Failed to connect to: %#v. %s", contact, err)
```

This way, you have to run through the whole method in order to understand that `dialog` is `nil`.
`return nil, error` is more readable than `return dialog, error`, when dialog is was not initialised before (as it was before change):
```go
...
return nil, fmt.Errorf("Failed to connect to: %#v. %s", contact, err)
```

Reply:
Returning `nil` is unsafe and is kinda agains Golang idealogies.
It's such a practice that method ALWAYS return that he promised to return

Reply:
This is exactly why higher layed can do tricks like:
```
dialogInstance, err := establisher.CreateDialog()
defer dialogInstance.Close()
```